packages/selenium-ide/src/neo/IO/filesystem.js

```
@@ -82,15 +82,17 @@ function downloadProject(project) {
       project.snapshot = snapshot
       Object.assign(project, Manager.emitDependencies())
     }
-    return browser.downloads.download({
-      filename: sanitizeProjectName(project.name) + '.side',
-      url: createBlob(
-        'application/json',
-        beautify(JSON.stringify(project), { indent_size: 2 })
-      ),
-      saveAs: true,
-      conflictAction: 'overwrite',
-    })
+    return UiState.isControlled
+      ? Promise.resolve()
```

This should not silently fail, we should send a message to the controller that the user requested to save.

Look at [PlaybackState](https://github.com/SeleniumHQ/selenium-ide/blob/v3/packages/selenium-ide/src/neo/stores/view/PlaybackState.js#L291) to get an idea of how to send an event over to an extension.

I was thinking of something like:
```js
{
      action: 'event',
      event: 'saveProject',
      options: {
        project
      }
}
```

---

We already send the plugin with the emit of the project entity. Do we expect to send another event here?

---

Emit project is something else, in controlled mode we shouldn't emit to begin with.

---

So to clarify. 
You want in control mode, IDE will send a saveProject event, do you still want to keep the emit? I guess we should still keep it for consistency between control and non-control mode?